src/diageig.jl

```diff
@@ -0,0 +1,30 @@
+import LinearAlgebra: eigvals, eigvecs, eigen, Diagonal
+using LinearAlgebra
+
+export _diageig, diageig_back, sym
+
+function _diageig(A)
+    sA = Symmetric(A)
+    eigval, eigvec = eigvals(sA), eigvecs(sA)
+    eigval, eigvec'
+end
+
+function sym(A)
+    (A + A') / 2.
+end
+
+function h(t; ϵ=0.00000001)
+    max(abs(t), ϵ)*sign(t)
+end
+
+function fij(λ)
```

Comment:
Try the following code snippet, it can be used to replace `h` and `fij` functions.
```julia
F = t .- t'
F .= F./(F.^2 .+ 1e-8)
```